src/MOI_wrapper.jl

```diff
@@ -1,58 +1,102 @@
-using LinQuadOptInterface
-
-const LQOI = LinQuadOptInterface
-const MOI  = LQOI.MOI
-
-const SUPPORTED_OBJECTIVES = [
-    LQOI.SinVar,
-    LQOI.Linear,
-    LQOI.Quad
-]
+import MathOptInterface
+
+const MOI = MathOptInterface
+
+include("CleverDict.jl")
+
+index_to_key(::Type{MOI.VariableIndex}, index::Int) = MOI.VariableIndex(index)
+key_to_index(key::MOI.VariableIndex) = key.value
+
+@enum(VariableType, CONTINUOUS, BINARY, INTEGER, SEMIINTEGER, SEMICONTINUOUS)
+@enum(BoundType, NONE, LESS_THAN, GREATER_THAN, LESS_AND_GREATER_THAN, INTERVAL, EQUAL_TO)
+@enum(ObjectiveType, SINGLE_VARIABLE, SCALAR_AFFINE, SCALAR_QUADRATIC)
+
+mutable struct VariableInfo
+    index::MOI.VariableIndex
+    column::Int
+    bound::BoundType
+    type::VariableType
+    name::String
+    # Storage for constraint names associated with variables because Gurobi
+    # can only store names for variables and proper constraints.
+    # We can perform an optimization and only store three strings for the
+    # constraint names because, at most, there can be three SingleVariable
+    # constraints, e.g., LessThan, GreaterThan, and Integer.
+    lessthan_name::String
+    greaterthan_interval_or_equalto_name::String
+    type_constraint_name::String
+    function VariableInfo()
+        return new(MOI.VariableIndex(-1), 0, NONE, CONTINUOUS, "", "", "", "")
```

Comment:
I would prefer VariableIndex(0) to signify an invalid one. Negative index means bridged. Even better would be to make the constructor with index and column arg